src/event-emitter.js

```diff
@@ -26,11 +26,18 @@ define(['lodash-amd/modern/arrays/pull',
   };
 
   EventEmitter.prototype.trigger = function (eventName, args) {
-    var listeners = this._listeners[eventName] || Immutable.Set();
 
-    listeners.forEach(function (listener) {
-      listener.apply(null, args);
-    });
+    //fire events like my:custom:event -> my:custom -> my
+    var events = eventName.split(':');
+    while(events.length){
+      var currentEvent = events.join(':');
+      var listeners = this._listeners[currentEvent] || Immutable.Set();
```

Comment:
I'm not sure this is quite right, I think the scope of this statement (due to the initialisation) is outside the while loop? We don't want the listeners to vary during the execution right?